### PR TITLE
Grafana/ui: Add Space component

### LINKED ISSUE #77185: grafana/ui: Add Space component
**Why is this component needed**:
Currently we have a few places where the components with spacing logic are defined: 
- [Space](https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/azuremonitor/components/Space.tsx) component in Grafana core.
- [Space](https://github.com/grafana/grafana-experimental/blob/main/src/QueryEditor/Space.tsx) component in the `@grafana/experimental` package, which is the exact copy of the `Space` component from Grafana.
- [Spacer](https://github.com/grafana/grafana/blob/main/public/app/features/alerting/unified/components/Spacer.tsx) component in Grafana core.

All these components should be unified into one component which is moved into `@grafana/ui` to reduce the code duplication.
___
 - [X] Is/could it be used in more than one place in Grafana?

**Implementation** (Checklist meant for the person implementing the component)

- [ ] Component has a story in Storybook.
- [ ] Props and naming follows [our style guide](https://github.com/grafana/grafana/blob/main/contribute/style-guides/frontend.md).
- [ ] It is extendable (rest props are spread, styles with className work, and so on).
- [ ] Uses [theme for spacing, colors, and so on](https://github.com/grafana/grafana/blob/main/contribute/style-guides/themes.md).
- [ ] Works with both light and dark theme.

**Documentation**

- [ ] Properties are documented.
- [ ] Use cases are described.
- [ ] Code examples for the different use cases.
- [ ] Dos and don'ts.
- [ ] Styling guidelines, specific color usage (if applicable).

### PR DESCRIPTION
**Pull request by Clarity-89**
**What is this feature?**

Add `Space` component to grafana/ui.

**Why do we need this feature?**

The same component is defined in core Grafana and grafana/experimental. This PR removes that duplication by moving the component to grafana/ui, from where it can be imported inside Grafana as well as plugins. 

**Who is this feature for?**

Grafana and plugin developers. 

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana/issues/77185

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: packages/grafana-ui/src/components/Layout/Space.mdx [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { Meta, ArgTypes } from '@storybook/blocks';
+import { Space } from './Space';
+
+<Meta title="MDX|Space" component={Space} />
+
+# Space
+
+The `Space` component is a wrapper around the `Box` component, used to add space between elements. Horizontal space is added using the `h` prop, while vertical space is added using the `v` prop.
```
Comment (joshhunt) [inline packages/grafana-ui/src/components/Layout/Space.mdx:8]: > is a wrapper around the `Box` component

I don't think you need to say this. This is an unimportant implementation detail.

**Review by joshhunt — APPROVED** [review]

File: packages/grafana-ui/src/components/Layout/Space.mdx [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import { Meta, ArgTypes } from '@storybook/blocks';
+import { Space } from './Space';
+
+<Meta title="MDX|Space" component={Space} />
+
+# Space
```
Comment (JoaoSilvaGrafana) [inline packages/grafana-ui/src/components/Layout/Space.mdx:6]: I'm wondering if we should have something here (or in Saga) explaining what the `layout` prop is for? Like when should you use `inline` or `block`?

File: packages/grafana-ui/src/components/Layout/Space.story.tsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+      <Space {...args} />
+      <Box borderStyle={'solid'} padding={1}>
+        Box without space
+      </Box>
+    </div>
+  );
+};
+
+Horizontal.args = {
+  v: 0,
```
Comment (JoaoSilvaGrafana) [inline packages/grafana-ui/src/components/Layout/Space.story.tsx:38]: Could we add some argTypes for these? You can use helpers like the ones in [Box](https://github.com/grafana/grafana/blob/main/packages/grafana-ui/src/components/Layout/Box/Box.story.tsx#L62). Then it makes it easier to see what options you can select.

**Review by JoaoSilvaGrafana — COMMENTED** [review]
Looks good, just leaving some small comments!

**Pull request merged**